cmd/main.go

```diff
@@ -1,109 +1,172 @@
+    ctx := context.Background()
+
+    slog.Info("Initializing PostgreSQL connection")
+    pg, err := postgres.New(ctx, config.Dsn)
+    if err != nil {
+        log.Fatalf("error connection to db: %s", err)
+    }
+    slog.Info("Connected to PostgreSQL")
+
+    h := handler.New(pg, config)
+    if err != nil {
+        log.Fatalf("error create handler: %s", err)
+    }
+
+    optsTele := tele.Settings{
+        Token:  config.Token,
+        Poller: &tele.LongPoller{Timeout: internal.Timeout},
+    }
+
+    bot, err := tele.NewBot(optsTele)
+    if err != nil {
+        log.Fatalf("error create bot instance: %s", err)
+    }
+    slog.Info("Bot instance created")
+
+    // getting information about profiles
+    uniqUserIds := map[int64]struct{}{}
+    for _, chatId := range config.AllowedChats {
+        // Convert old group ID to supergroup ID if needed
+        if chatId > 0 {
+            chatId = -1000000000000 + chatId
+        }
+
+        slog.Debug("Fetching profile IDs for chat", "chatId", chatId)
+        profileIds, err := pg.GetProfileIdsByChatId(ctx, chatId)
+        if err != nil {
+            slog.Warn("failed to get profile ids for chat", "chatId", chatId, "error", err)
+            continue
+        }
+
+        for _, id := range profileIds {
+            if _, ok := uniqUserIds[id]; !ok {
+                uniqUserIds[id] = struct{}{}
+            } else {
+                continue
+            }
+
+            profile, err := bot.ChatMemberOf(tele.ChatID(chatId), &tele.User{ID: id})
+            if err != nil {
+                slog.Warn("failed to get profile info for id", "id", id, "error", err)
+                continue
+            }
+
+            p := model.NewProfile(profile.User)
+            if err := pg.StoreProfile(ctx, p); err != nil {
+                slog.Warn("failed to store profile with id", "id", profile.User.ID, "error", err)
+            } else {
+                slog.Debug("Stored profile with id", "id", profile.User.ID)
+            }
+        }
+    }
+    uniqUserIds = nil
+
+    // Commands handlers
+    bot.Handle(internal.HelpCmd, h.Help)
+    bot.Handle(internal.HCmd, h.Help)
+    bot.Handle(internal.StartCmd, h.Help)
+    bot.Handle(internal.VerCmd, h.Ver)
+    bot.Handle(internal.VCmd, h.Ver)
+
+    bot.Handle(internal.TopCmd, h.GetTop)
+    bot.Handle(internal.BottomCmd, h.GetBottom)
+    bot.Handle(internal.ProfileCmd, h.GetProfileCount)
+    bot.Handle(internal.TopicCmd, h.SetTopic)
+    bot.Handle(internal.EventCmd, h.EventCmd)
+    bot.Handle(internal.TodayCmd, h.TodayCmd)
+
+    // Button handlers
+    bot.Handle("\f"+internal.ShareBtn, h.EventCallback)
+
+    // Handle only messages in allowed groups (msg.Chat.Type = "group" | "supergroup")
+    // private messages handles only by command endpoint handler
+    bot.Handle(tele.OnText, func(c tele.Context) error {
+        h.Count(c)
+        slog.Debug("Received text message", "message", c.Message().Text)
+        respond := false
```

Comment:
all this code up to string 161 must be in handler/chatgpt.go file and wrapped to function also.
Here should be something like:
```
        bot.Handle(tele.OnText, func(c tele.Context) error {
            h.Count(c)
            h.CGPTRequest(c) // <--- here call you funtion on handler/chargpt.go, function getting all contect in c. Name the function how you want, it is just a suggestion
        })
```